archive/extract_test.go

```diff
@@ -129,6 +129,9 @@ func testExtract(t *testing.T, when spec.G, it spec.S) {
 		})
 
 		it("extracts a tar file", func() {
+			old := archive.SetUmask(0)
```

Comment:
Is there a test we can write to not regress on expected behavior? I'm not entirely sure what happens if we leave off these 2 lines, for instance. Will this test fail and will it be clear why it failed? Should we be asserting the value of `archive.SetUmask(0)` to be `old` at the end of these tests?

Reply:
If we leave off these two lines then the test will fail with `umask should be unset by the calling function`.

Reply:
Can we write a test for that so we don't accidentally remove that error behavior in some distant future?

Reply:
Makes sense, added in c0ed744